gpt_index/indices/base.py

```diff
@@ -86,6 +87,7 @@ def __init__(
         self._prompt_helper = prompt_helper or PromptHelper.from_llm_predictor(
             self._llm_predictor, chunk_size_limit=chunk_size_limit
         )
+        self._text_splitter = text_splitter or TokenTextSplitter()
```

Comment:
if this is true, then wouldn't all the expressions of 
```
text_splitter = (
            self._text_splitter
            or self._prompt_helper.get_text_splitter_given_prompt(prompt_with_schema, 1)
        )
```
always evaluate to using this text splitter and never the prompt helper?

Reply:
I believe all cases of this are in classes that don't inherit from `BaseGPTIndex`? For example, `SQLContextBuilder`.

In the case of `GPTTreeIndex`, it's overwriting the initial value set by the parent. Lmk if you see a mistake there.